app/scripts/models/flow.js

```diff
@@ -11,79 +11,84 @@
  * If unavailable there, fetch from data attributes in the DOM.
  */
 
-define(function (require, exports, module) {
-  'use strict';
 
-  const $ = require('jquery');
-  const AuthErrors = require('../lib/auth-errors');
-  const Backbone = require('backbone');
-  const Cocktail = require('cocktail');
-  const ErrorUtils = require('../lib/error-utils');
-  const ResumeTokenMixin = require('./mixins/resume-token');
-  const SearchParamMixin = require('./mixins/search-param');
-  const vat = require('../lib/vat');
+const $ = require('jquery');
+const AuthErrors = require('../lib/auth-errors');
+const Backbone = require('backbone');
+const Cocktail = require('cocktail');
+const ErrorUtils = require('../lib/error-utils');
+const ResumeTokenMixin = require('./mixins/resume-token');
+const SearchParamMixin = require('./mixins/search-param');
+const vat = require('../lib/vat');
+const Url = require('../lib/url');
 
-  var Model = Backbone.Model.extend({
-    initialize (options) {
-      options = options || {};
+var Model = Backbone.Model.extend({
+  initialize (options) {
+    options = options || {};
 
-      this.sentryMetrics = options.sentryMetrics;
-      this.window = options.window || window;
-
-      // We should either get both fields from the resume token, or neither.
-      // Getting one without the other is an error.
-      this.populateFromStringifiedResumeToken(this.getSearchParam('resume'));
-      if (this.has('flowId')) {
-        if (! this.has('flowBegin')) {
-          this.logError(AuthErrors.toMissingResumeTokenPropertyError('flowBegin'));
-        }
-      } else if (this.has('flowBegin')) {
-        this.logError(AuthErrors.toMissingResumeTokenPropertyError('flowId'));
-      } else {
-        this.populateFromDataAttribute('flowId');
-        this.populateFromDataAttribute('flowBegin');
+    this.sentryMetrics = options.sentryMetrics;
+    this.window = options.window || window;
+    this.metrics = options.metrics;
+    const urlParams = Url.searchParams(this.window.location.search);
+    // We should either get both fields from the resume token, or neither.
+    // Getting one without the other is an error.
+    this.populateFromStringifiedResumeToken(this.getSearchParam('resume'));
+    if (this.has('flowId')) {
+      if (! this.has('flowBegin')) {
+        this.logError(AuthErrors.toMissingResumeTokenPropertyError('flowBegin'));
       }
-    },
+    } else if (this.has('flowBegin')) {
+      this.logError(AuthErrors.toMissingResumeTokenPropertyError('flowId'));
+    } else if (urlParams.flow_begin_time && urlParams.flow_id) {
+      this.set('flowId', urlParams.flow_id);
```

Comment:
Should flow_begin_time and flow_id be checked for validity? The [schema below](https://github.com/mozilla/fxa-content-server/pull/6227/files?w=1#diff-0a6ffa0f4795903fe29afcfc1e11ff95R80) is even used when populating from the data attributes.